macros/src/lib.rs

```diff
@@ -351,7 +351,7 @@ pub fn exception(args: TokenStream, input: TokenStream) -> TokenStream {
             let tramp_ident = Ident::new(&format!("{}_trampoline", f.sig.ident), Span::call_site());
             let ident = &f.sig.ident;
 
-            if let Some(error) = check_for_blacklisted_attrs(&f.attrs) {
+            if let Err(error) = check_attr_whitelist(&f.attrs) {
```

Comment:
Does this need to be duplicated into every match arm or can be move before the match?

Reply:
Not sure I follow what you are asking here. Could you elaborate?

Reply:
`exception` has 3 cases (`DefaultHandler`, `HardFault`, and "everything else"), and right now they all call `check_attr_whitelist`. I think it would be less duplication if there was just one call to it at the beginning of the `exception` function.

Reply:
Ah right, fixed it